experiments/tf_trainer/keras_gru_attention/run.py

```diff
@@ -21,6 +21,8 @@
 tf.app.flags.DEFINE_string("embeddings_path",
                            "local_data/glove.6B/glove.6B.100d.txt",
                            "Path to the embeddings file.")
+tf.app.flags.DEFINE_string('key_name', 'comment_key',
```

Comment:
I oppose defining flags in more than one place. They are global variables (and thus pretty bad.)
My suggestion is that this properly belongs where it is, and that you expose access to it by making it a member of model_trainer (self._key_name and add an accessor.)

I think serving_input should be removed entirely and its functionality moved into model_trainer. This might also help.

Reply:
Sorry, flags that *share the same name* should be defined once and only once globally. I don't really care where, but probably closest to where they are used the most.

Another good principle is that flags should scope only to the file, and only to the main() function within the file (which is itself global).

Reply:
In this case, in this PR, the flag redundancy is removed right? It is defined in the run.py and is used only there. Am I missing anything?